picard/util/__init__.py

```diff
@@ -708,3 +709,13 @@ def extract_year_from_date(dt):
             return parse(dt).year
     except (TypeError, ValueError):
         return None
+
+
+def cached_settings(keys, settings=None, config=None):
+    if settings is None:
+        settings = {}
```

Comment:
What is the difference between:
```
def cached_settings(keys, settings=None, config=None):
    if settings is None:
        settings = {}
```
and
```
def cached_settings(keys, settings={}, config=None):
```
except that the latter has 2 fewer lines of code and more clearly defines what type of object should be passed as settings.

Reply:
In the second case all calls to cached_settings without a settings parameter share the same object (because the {} as default gets created only once when the function is declared):

```python
s1 = cached_settings(['write_id3v23'])
print(s)  # e.g. {'write_id3v23': True}

s2 = cached_settings(['windows_compatibility'])
print(s)  # e.g. {'write_id3v23': True, 'windows_compatibility': False}

s1 == s2  # True
```

That could cause unwanted bugs if unrelated calls to this function share the same object.

Reply:
Ah - I didn't realise that the default values were static!!! I bet that is a common gotcha.

Reply:
It is indeed. I think our linting on codacy would have complained loud and clear about this. There might be some rare cases where this behavior makes sense, but due to its surprising nature the general recommendation is to avoid this.